modules/git/git.go

```diff
@@ -286,6 +286,15 @@ func syncGitConfig() (err error) {
 		}
 	}
 
+	if setting.Git.HTTPFollowRedirects == "" {
+		setting.Git.HTTPFollowRedirects = "false"
+	}
+
+	// http.followRedirects defaults to initial however this could still be abused by migration endpoints -- therefore disable this
+	if err := configSet("http.followRedirects", setting.Git.HTTPFollowRedirects); err != nil {
```

Comment:
```suggestion
	// http.followRedirects defaults to initial however this could still be abused by migration endpoints -- therefore disable this
	followHTTPRedirects, _ := strconv.ParseBool(setting.Git.HTTPFollowRedirects)
	if err := configSet("http.followRedirects", strconv.FormatBool(followHTTPRedirects)); err != nil {
```
Might be a little bit overhead, but much more user friendly.

Reply:
Oh, additionally this one will always behave correctly, while I don't know what Git's behavior would be when passing i.e. `a-non-bool-term` as value.
Would it die, or would it ignore the invalid value?

Reply:
It's  not a bool, it can take `initial` as well, which happens to be the default:

https://git-scm.com/docs/git-config/2.15.4#Documentation/git-config.txt-httpfollowRedirects

Reply:
It is not a bool.